feat(SelectE): Add component

thyhjwb6 opened the pull request:
## Related issue
Closes #2843 

## Overview
Adds `SelectE` with basic functionality.

## Link to preview
https://select-e.netlify.app/?path=/story/select-experimental--default

## Reason
Required for v3

## Work carried out
- [x] Refactor existing code
- [x] Add component
- [x] Add basic functionality

## Screenshot
![2021-11-29 14 10 59](https://user-images.githubusercontent.com/56078793/143882883-5111fb98-0706-4e80-98a9-b2886faee039.gif)

## Developer notes
This does not include:
- clear button
- badge/number
- icons
- other potential refactors based on reuse
- correct colour for item hover

thyhjwb6 committed:
refactor(TextInput): Expose sizes

thyhjwb6 committed:
feat(SelectE): Add component

thyhjwb6 committed:
feat(SelectE): Add ability to disable

thyhjwb6 committed:
feat(SelectE): Add error state

thyhjwb6 committed:
feat(SelectE): Add ability to set value

m7kvqbe1 reviewed (changes requested):
Looks good! Much nicer than the old implementation!

Appreciate that it's early days but couple of visual things I noticed in Firefox:

No outer light box-shadow on the sheet:
<img width="36" alt="Screenshot 2021-11-29 at 18 01 09" src="https://user-images.githubusercontent.com/48086589/143919864-26420a78-b27b-47f7-9b26-01a6290e75d6.png">

Grey hover state of bottom option corner clips the container:
<img width="86" alt="Screenshot 2021-11-29 at 18 01 22" src="https://user-images.githubusercontent.com/48086589/143919906-d5af57d1-2534-4e12-88d2-0aa64972c75a.png">

Should this button have a pointer cursor on hover:

<img width="50" alt="Screenshot 2021-11-29 at 18 01 44" src="https://user-images.githubusercontent.com/48086589/143919965-017e02cf-61eb-4d06-970f-0d223164ce0d.png">

thyhjwb6 committed:
fix(NumberInputE): Add pointer to buttons

thyhjwb6 committed:
refactor(OuterWrapper): Remove negative margins

These were causing consequences which were
difficult to develop other components with.
